server/api/lib/tags/index.js

```diff
@@ -1,25 +1,39 @@
-
 const Exception = require('../../../api/model/exception');
-const functions = [require('../tags/housing'),require('../tags/public')];
-
-const generateTagsForPlan = async (planId) => {
-	const planTags = [];
-	for (const counter in functions) {
-		const tagFunction = functions[counter];
-		try {
-			if (typeof(tagFunction.doesTagApply) != "function") {
-				throw new Exception.BadRequest(`"Please require only doesTagApply functions.`);
-			};
-			const result = await tagFunction.doesTagApply(planId);
-			planTags.push(...result);
-		} catch (err) {
-			console.debug(err);			
+const { getTagsResources } = require('../tags/tags_resources');
+
+const housingTag = require('../tags/housing');
+const publicBuildingsTag = require('../tags/public');
+const forestTag = require('../tags/forests');
+const streamTag = require('./streams');
+const lightRailTag = require('./light_rail');
+
+const taggingFunctions = [
+	housingTag,
+	publicBuildingsTag,
+	forestTag,
+	streamTag,
+	lightRailTag
+];
+
+
+const getPlanTagger = async () => {
+	const resources = await getTagsResources();
+
+	return async (plan) => {
+		const planTags = [];
+		for (const tagFunction of taggingFunctions) {
+			const result = await tagFunction.doesTagApply(plan, resources);
+
+			if (result !== null) {
+				planTags.push(result);
+			}
+
 		}
-			
-	} 
-	return planTags;	
-}
+		return planTags;
+	};
+};
+
 
 module.exports = {
-	generateTagsForPlan
+	getPlanTagger
```

Comment:
Prefereably renamed to `PlanTagger` or without a name, by using the following syntax
```
module.exports  = PlanTagger;
```

Then, when requiring it, it can be given any name:

```
const PlanTagger = require('../lib/tags');
const Tagger = require('../lib/tags'); // or any other name
```

Reply:
It's not a class, it's a function that returns the plan tagger. I'll make the export as you stated, but in my opinion it still should be called "getPlanTagger". After you invoke it, you get the plan tagger.

Reply:
Hi, it's a common js practive - you just require functions, without needing to invoke. The class in js is a function.